### DISCUSSION TITLE
Use of chunkReason in webpack SplitChunksPlugin

### DISCUSSION THREAD

**Discussion by karthickshanmugam0689** [webpack/webpack#18019] [Q&A]
As of now, Im thinking to use `WebpackManifestPlugin` to generate json file, which I would use to determine the js files to prefetch before even actually using the application.

This works good but in this json file, I would like to avoid the files that are lazily imported in application(via `import()`). As these files are anyway lazily imported, I might not need them on demand and hence can be delayed or avoided to be prefetched.
I was going through the chunks and found that each and every chunk has some field called `chunkReason`. Majorly I saw that `chunkReason` seems to be `split chunk (cache group: defaultVendors)` [in case of lazily imported chunks](https://github.com/webpack/webpack/blob/main/lib/optimize/SplitChunksPlugin.js#L1524).
Can i fairly assume that I can use `chunkReason` as an identifier to separate lazy loaded chunks? Or is it possible to have any identifier from `Chunk` to let me know if it is lazily loaded?

**Comment by alexander-akait** [comment]
You can use this https://webpack.js.org/plugins/split-chunks-plugin/#splitchunkschunks

**Comment by karthickshanmugam0689** [comment]
Would you mean to include cacheGroup for these async chunks with `splitChunks.[cacheGroup].chunks: 'async'`?

**Reply by alexander-akait** [reply]
Yes, you can merge chunks into one
